AlphaWallet/Tokens/ViewControllers/NewTokenViewController.swift

```diff
@@ -314,20 +325,28 @@ class NewTokenViewController: UIViewController {
         let symbol = symbolTextField.value
         let decimals = Int(decimalsTextField.value) ?? 0
         guard let tokenType = self.tokenType else { return }
-        //TODO looks wrong to mention ERC875TokenBalance specifically
-        var balance: [String] = viewModel.erc875TokenBalance
+
+        var nonFungibleBalance: NonFungibleBalance
+        switch tokenType {
+        case .erc875:
+            if let balance = viewModel.nonFungibleBalance {
+                nonFungibleBalance = balance.rawValue.isEmpty ? .erc875(["0"]) : balance
+            } else {
+                nonFungibleBalance = .erc875(["0"])
+            }
+        case .nativeCryptocurrency, .erc20, .erc721, .erc1155:
+            nonFungibleBalance = viewModel.nonFungibleBalance ?? .balance([])
```

Comment:
This one's an impossible path right? Shall we do an `assertImpossibleCodePath(message:)`? 

(we'll change the implementation for `assertImpossibleCodePath(message:)` and `assertImpossibleCodePath()` to use `preconditionFailure(_:)` instead of `assert(Bool,String)` in the future)

Reply:
i guess for that we need to tweak ErcToken model, because nonFungible balance isn't optional, and in any case balance need to be calculated